cli/filplus.go

```diff
@@ -923,3 +931,448 @@ var filplusSignRemoveDataCapProposal = &cli.Command{
+			if err != nil {
+				return nil, fmt.Errorf("could not load the claim %d: %w", c, err)
+			}
+			if !ok {
+				return nil, fmt.Errorf("claim %d not found in the actor state", c)
+			}
+			if claim.TermMax < tmax && claim.TermStart+claim.TermMax > head.Height() {
+				// If client is not same - needs to burn datacap
+				if claim.Client != wid {
+					newClaims = append(newClaims, verifregtypes9.ClaimExtensionRequest{
+						Claim:    c,
+						Provider: prov.addr,
+						TermMax:  tmax,
+					})
+					rDataCap.Add(big.NewInt(int64(claim.Size)).Int, rDataCap.Int)
+					continue
+				}
+				terms = append(terms, verifregtypes9.ClaimTerm{
+					ClaimId:  c,
+					TermMax:  tmax,
+					Provider: prov.id,
+				})
+			}
+		}
+	}
+
+	var msgs []*types.Message
+
+	if len(terms) > 0 {
+		params, err := actors.SerializeParams(&verifregtypes9.ExtendClaimTermsParams{
+			Terms: terms,
+		})
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to searialise the parameters: %w", err)
+		}
+
+		oclaimMsg := &types.Message{
+			To:     verifreg.Address,
+			From:   wallet,
+			Method: verifreg.Methods.ExtendClaimTerms,
+			Params: params,
+		}
+
+		msgs = append(msgs, oclaimMsg)
+	}
+
+	if len(newClaims) > 0 {
+		// Get datacap balance
+		aDataCap, err := api.StateVerifiedClientStatus(ctx, wallet, types.EmptyTSK)
+		if err != nil {
+			return nil, err
+		}
+
+		if aDataCap == nil {
+			return nil, fmt.Errorf("wallet %s does not have any datacap", wallet)
+		}
+
+		// Check that we have enough data cap to make the allocation
+		if rDataCap.GreaterThan(big.NewInt(aDataCap.Int64())) {
+			return nil, fmt.Errorf("requested datacap %s is greater then the available datacap %s", rDataCap, aDataCap)
+		}
+
+		ncparams, err := actors.SerializeParams(&verifregtypes9.AllocationRequests{
+			Extensions: newClaims,
+		})
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to searialise the parameters: %w", err)
+		}
+
+		transferParams, err := actors.SerializeParams(&datacap2.TransferParams{
+			To:           builtin.VerifiedRegistryActorAddr,
+			Amount:       big.Mul(rDataCap, builtin.TokenPrecision),
+			OperatorData: ncparams,
+		})
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to serialize transfer parameters: %w", err)
+		}
+
+		nclaimMsg := &types.Message{
+			To:     builtin.DatacapActorAddr,
+			From:   wallet,
+			Method: datacap.Methods.TransferExported,
+			Params: transferParams,
+			Value:  big.Zero(),
+		}
+
+		fmt.Printf("Some if the specified allocation have a different client address and will require %d Datacap to extend\n", rDataCap.Int)
+
+		if assumeYes {
+			var yes bool
+			yes, err = confirm(ctx)
+			if err != nil {
+				return nil, fmt.Errorf("error reading user input: %w", err)
+			}
+			if !yes {
+				fmt.Printf("Dropping the extension for claims that require Datacap")
+				return msgs, nil
+			}
+		}
+
+		msgs = append(msgs, nclaimMsg)
+	}
+
+	return msgs, nil
+}
+
+func confirm(ctx context.Context) (bool, error) {
```

Comment:
Consider instead of this:
input, err = (&promptui.Prompt{Label: d.T("Proceed? Yes [y] / No [default], Ctrl+C (^C) to exit")}).Run()
if err != nil || !(strings.EqualFold(input, "y") || strings.EqualFold(input, "yes") {
    log.Fatal("Exiting on prompt's request")
}